src/v/cluster/tests/serialization_rt_test.cc

```diff
@@ -458,8 +458,23 @@ struct adl<partition_status_v1> {
 } // namespace reflection
 
 template<typename T>
-void roundtrip_test(const T original) {
-    auto serde_in = original;
+std::pair<T, T> compat_copy(T t) {
+    return {t, t};
+}
+
+template<typename T>
+concept ExplicitCopyable = requires(T a) {
+    { a.copy() } -> std::same_as<T>;
+};
+
+template<ExplicitCopyable T>
+std::pair<T, T> compat_copy(T t) {
+    return {t.copy(), std::move(t)};
+}
+
+template<typename T>
+void roundtrip_test(T a) {
+    auto&& [serde_in, original] = compat_copy(std::move(a));
```

Comment:
The implementation of `compat::compat_copy` can be used here, whilst still allowing alternative implementations via ADL, should that be required (as in the suggestion), or `compat_copy` can be qualified.

Taking rvalue ref to the result of `compat_copy` is a bit confusing here, because although it is valid and produces the expected result, it begs the question as to whether there can be any aliasing of the two returned values or the original, which would be an unexpected result for a copy.
```suggestion
template<typename T>
void roundtrip_test(T a) {
    using compat::compat_copy;
    auto [serde_in, original] = compat_copy(std::move(a));
```